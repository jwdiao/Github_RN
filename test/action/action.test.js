import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import request from '../__mocks__/request'
jest.mock('../__mocks__/request');
const middlewares = [thunk] // add your middlewares like `redux-thunk`
const mockStore = configureStore(middlewares)

// You would import the action from your codebase in a real scenario
function success(data) {
  return {
    type: 'FAVORITE_LOAD_SUCCESS',
    projectModels: data
  }
}

function fetchData () {
  return dispatch => {
    return request() // Some async action with promise
      .then((data) => dispatch(success(data)))
  };
}

it('should execute fetch data', () => {
  const store = mockStore({})

  // Return the promise
  return store.dispatch(fetchData())
    .then(() => {
      const actions = store.getActions()
      expect(actions[0]).toMatchSnapshot()
    })
})
