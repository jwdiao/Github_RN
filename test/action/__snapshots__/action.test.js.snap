// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should execute fetch data 1`] = `
Object {
  "projectModels": Array [
    Object {
      "projectModel": Object {
        "isFavorite": true,
        "item": Object {
          "description": "The open source codebase and curriculum. Learn to code for free together with millions of people.",
        },
      },
    },
    Object {
      "projectModel": Object {
        "isFavorite": true,
        "item": Object {
          "description": "JavaScript Style Guide",
        },
      },
    },
    Object {
      "projectModel": Object {
        "isFavorite": true,
        "item": Object {
          "description": "技术面试必备基础知识",
        },
      },
    },
  ],
  "type": "FAVORITE_LOAD_SUCCESS",
}
`;
